bundles/org.openhab.binding.ism8/src/main/java/org/openhab/binding/ism8/internal/Ism8Handler.java

```diff
@@ -0,0 +1,170 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.ism8.internal;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.library.types.QuantityType;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.ism8.server.DataPointChangedEvent;
+import org.openhab.binding.ism8.server.IDataPoint;
+import org.openhab.binding.ism8.server.IDataPointChangeListener;
+import org.openhab.binding.ism8.server.Server;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link Ism8Handler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Hans-Reiner Hoffmann - Initial contribution
+ */
+@NonNullByDefault
+public class Ism8Handler extends BaseThingHandler implements IDataPointChangeListener {
+    private final Logger logger = LoggerFactory.getLogger(Ism8Handler.class);
+
+    private @Nullable Ism8Configuration config;
+    private @Nullable Server server;
+
+    public Ism8Handler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("Ism8: Handle command = {} {}", channelUID.getId(), command);
+        if (isLinked(channelUID)) {
+            Channel channel = null;
+            for (Channel ch : getThing().getChannels()) {
+                if (ch.getUID().getId().equals(channelUID.getId())) {
+                    channel = ch;
+                    break;
+                }
+            }
```

Comment:
OK, but what happen if an ID does not exist? Does you suggestion return a 'null' in case the ID doesn't exist?

Reply:
After testing all the changes I had to use the original code as this change was resulting in an exception. The initial version does not load channel in case nothing is found. After this block is the channel checked against 'null', so nothing can happen. The suggestion does fail in some cases. Do we need to wrap this around an try/catch or can the original code be used?

Reply:
After testing the binding I had to return to the original version as the simplification throwed an exception. The original version does simply result in having the channel = 'null' and this is checked afterwards. Wrapping this into a try/catch isn't the best choice right?

Reply:
` getThing().getChannel(channelUID)` returns null if the channel wasn't found, exactly as your current implementation. As  `getThing().getChannel(channelUID)` doesn't throw any checked exceptions, a try/catch block is not needed. Can you be more specific what exactly doesn't work?

Reply:
I noticed that any write command wasn't send to the heating system. After checking the log file I saw an error thrown exactly in this line (52). The log content was:

> 2020-07-28 00:39:48.490 [ERROR] [nal.common.AbstractInvocationHandler] - An error occurred while calling method 'ThingHandler.handleCommand()' on 'org.openhab.binding.ism8.internal.Ism8Handler@1af8720': org.eclipse.smarthome.core.thing.Thing.getChannel(Lorg/eclipse/smarthome/core/thing/ChannelUID;)Lorg/eclipse/smarthome/core/thing/Channel;
java.lang.NoSuchMethodError: org.eclipse.smarthome.core.thing.Thing.getChannel(Lorg/eclipse/smarthome/core/thing/ChannelUID;)Lorg/eclipse/smarthome/core/thing/Channel;
	at org.openhab.binding.ism8.internal.Ism8Handler.handleCommand(Ism8Handler.java:52) ~[?:?]
	at sun.reflect.GeneratedMethodAccessor85.invoke(Unknown Source) ~[?:?]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:?]
	at org.eclipse.smarthome.core.internal.common.AbstractInvocationHandler.invokeDirect(AbstractInvocationHandler.java:153) [102:org.eclipse.smarthome.core:0.10.0.oh240]
	at org.eclipse.smarthome.core.internal.common.InvocationHandlerSync.invoke(InvocationHandlerSync.java:59) [102:org.eclipse.smarthome.core:0.10.0.oh240]
	at com.sun.proxy.$Proxy145.handleCommand(Unknown Source) [253:org.openhab.binding.ism8:2.5.8.202007272221]
	at org.eclipse.smarthome.core.thing.internal.profiles.ProfileCallbackImpl.handleCommand(ProfileCallbackImpl.java:75) [109:org.eclipse.smarthome.core.thing:0.10.0.oh240]
	at org.eclipse.smarthome.core.thing.internal.profiles.SystemDefaultProfile.onCommandFromItem(SystemDefaultProfile.java:49) [109:org.eclipse.smarthome.core.thing:0.10.0.oh240]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:?]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:?]
	at org.eclipse.smarthome.core.internal.common.AbstractInvocationHandler.invokeDirect(AbstractInvocationHandler.java:153) [102:org.eclipse.smarthome.core:0.10.0.oh240]
	at org.eclipse.smarthome.core.internal.common.Invocation.call(Invocation.java:53) [102:org.eclipse.smarthome.core:0.10.0.oh240]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:?]
	at java.lang.Thread.run(Thread.java:748) [?:?]

It looks like the method doesn't exist, but compiling was going well. Is this method newly introduced in the OpenHab version? You may need additional to know that the heating system implementation is running on a Synology, but in version 2.4 only. Might this be a reason?

Reply:
That might be the reason. Would be good if you test your binding with 2.5, too.

Reply:
In the moment I started to create this binding version 2.4 was running quite stable and for this reason I did not update it (yet). Anyhow, I'll do it and check if everything works with version 2.5, but this rises the question if the code should not be kept as this would also work with 2.4 instead of only used in 2.5. Is there the need to somehow add an attribute to check a minimum OpenHab version? In this case someone would not be running into an issue when trying to load this in an older version.

Reply:
Good question. I don't think that your binding is offered to someone who runs on 2.4 via the UI.

Reply:
I have updated my OH-RT to V2.5.2 and the initial suggestion works now fine. So it looks that this method was introduced in V2.5. I have replaced my initial code.
Do you know if there is a kind of dependency attribute to ensure a minimum RT version or should we simply ignore this? Generally would it be good to have such check at moment a binding is loaded/activated instead of guessing in case something goes wrong at any later stage.

Reply:
I'm not aware of such a dependency check. But I think it isn't necessary at all, since I guess the normal user can only force to install you binding with 2.4 by moving your JAR into the addons folder. I don't think that your 2.5 add-on will be visible to 2.4 users in the install UI.